hs_geo_raster_resource/migrations/custom_migration_for_raster_meta_update_20160512.py

```diff
@@ -0,0 +1,143 @@
+                ori_dir = os.getcwd()
+                os.chdir(temp_dir)
+                vrt_file_path = [os.path.join(temp_dir, f) for f in os.listdir(temp_dir) if '.vrt' == os.path.splitext(f)[1]].pop()
+                res_md_dict = raster_meta_extract.get_raster_meta_dict(vrt_file_path)
+                os.chdir(ori_dir)
+                shutil.rmtree(temp_dir)
+
+                # update band metadata of raster resources
+                is_modified = False
+                if res_md_dict['band_info']:
+                    for i, band_meta in res_md_dict['band_info'].items():
+                        band_obj = res.metadata.bandInformation.filter(name='Band_{}'.format(str(i))).first()
+                        if band_obj:
+                            res.metadata.update_element('bandInformation',
+                                                        band_obj.id,
+                                                        maximumValue=band_meta['maximumValue'],
+                                                        minimumValue=band_meta['minimumValue'],
+                                                        noDataValue=band_meta['noDataValue'],
+                                                        )
+                            is_modified = True
+
+                # resource modify update
+                if is_modified:
+                    bag_name = 'bags/{res_id}.zip'.format(res_id=res.short_id)
+                    if istorage.exists(bag_name):
+                        # delete the resource bag as the old bag is not valid
+                        istorage.delete(bag_name)
+                    resource_modified(res, res.creator)
+
+                    success += 1
+                    log.info('Raster metadata is updated successfully for resource:ID:{} '
+                             'Title:{}'.format(res.short_id, res.metadata.title.value))
+                else:
+                    log.info('Raster metadata is not updated for resource:ID:{} '
+                             'Title:{}'.format(res.short_id, res.metadata.title.value))
+                    not_modify.append({res.short_id: res.metadata.title.value})
+
+            except Exception as e:
+                fail.append({res.short_id : res.metadata.title.value})
+                log.info('Raster metadata update failed for resource:ID:{} '
+                        'Title:{}'.format(res.short_id, res.metadata.title.value))
+
+    print 'total resources: {}'.format(str(RasterResource.objects.all().count()))
+    print 'success update resources: {}'.format(str(success))
+    print 'failed update resources: {}'.format(str(fail))
+    print 'not updated resources: {}'.format(str(not_modify))
+
+
+def undo_migrate_tif_file(apps, schema_editor):
+
+    # loop through all the band information and change the nodata value, min, max values as none
+    for res in RasterResource.objects.all():
+        try:
+            for band_obj in res.metadata.bandInformation:
```

Comment:
@jbradberry Good point. @gantian127 Also, I think it is better to at least print out exception error message for diagnosing and debugging purposes; otherwise, we have no clue whether undo migration has succeeded without errors or not.